RecoTracker/MkFit/plugins/MkFitIterationConfigESProducer.cc

```diff
@@ -6,35 +6,107 @@
 #include "RecoTracker/MkFit/interface/MkFitGeometry.h"
 
 // mkFit includes
+#include "Track.h"
+#include "TrackerInfo.h"
+#include "mkFit/HitStructures.h"
 #include "mkFit/IterationConfig.h"
 
+namespace {
+  using namespace mkfit;
+
+  void partitionSeeds0(const TrackerInfo &trk_info,
+                       const TrackVec &in_seeds,
+                       const EventOfHits &eoh,
+                       IterationSeedPartition &part) {
+    const int size = in_seeds.size();
+
+    for (int i = 0; i < size; ++i) {
+      const Track &S = in_seeds[i];
+
+      const bool z_dir_pos = S.pz() > 0;
+
+      HitOnTrack hot = S.getLastHitOnTrack();
+      float eta = eoh[hot.layer].GetHit(hot.index).eta();
+
+      // Region to be defined by propagation / intersection tests
+      TrackerInfo::EtaRegion reg;
+
+      const LayerInfo &outer_brl = trk_info.outer_barrel_layer();
+
+      const LayerInfo &tib1 = trk_info.m_layers[4];
+      const LayerInfo &tob1 = trk_info.m_layers[10];
+
+      const LayerInfo &tecp1 = trk_info.m_layers[27];
+      const LayerInfo &tecn1 = trk_info.m_layers[54];
+
+      const LayerInfo &tec_first = z_dir_pos ? tecp1 : tecn1;
+
+      float maxR = S.maxReachRadius();
+      float z_at_maxr;
+
+      bool can_reach_outer_brl = S.canReachRadius(outer_brl.m_rout);
+      float z_at_outer_brl;
+      bool misses_first_tec;
+      if (can_reach_outer_brl) {
+        z_at_outer_brl = S.zAtR(outer_brl.m_rout);
+        if (z_dir_pos)
+          misses_first_tec = z_at_outer_brl < tec_first.m_zmin;
+        else
+          misses_first_tec = z_at_outer_brl > tec_first.m_zmax;
+      } else {
+        z_at_maxr = S.zAtR(maxR);
+        if (z_dir_pos)
+          misses_first_tec = z_at_maxr < tec_first.m_zmin;
+        else
+          misses_first_tec = z_at_maxr > tec_first.m_zmax;
+      }
+
+      if (misses_first_tec) {
+        reg = TrackerInfo::Reg_Barrel;
+      } else {
+        if ((S.canReachRadius(tib1.m_rin) && tib1.is_within_z_limits(S.zAtR(tib1.m_rin))) ||
+            (S.canReachRadius(tob1.m_rin) && tob1.is_within_z_limits(S.zAtR(tob1.m_rin)))) {
+          reg = z_dir_pos ? TrackerInfo::Reg_Transition_Pos : TrackerInfo::Reg_Transition_Neg;
+        } else {
+          reg = z_dir_pos ? TrackerInfo::Reg_Endcap_Pos : TrackerInfo::Reg_Endcap_Neg;
+        }
+      }
+
+      part.m_region[i] = reg;
+      part.m_sort_score[i] = 5.0f * (reg - 2) + eta;
```

Comment:
perhaps these magic numbers need to be put into a constexpr or clarified via a comment?